Feature/cdh 10 16 release (#747)
SoftwareAG/c8y-docs: content/datahub/working-with-datahub-bundle/offloading-base-collections.md

### Offloading the base collections

> The following tables summarize the resulting schemas for each of the {{< product-c8y-iot >}} base collections. These schemas additionally include virtual columns *dir0*, ..., *dir3*, which are used for internal purposes. The columns are generated during the extraction process, but neither do they have corresponding data in the Operational Store of {{< product-c8y-iot >}}, nor are they persisted in the data lake. You must not use *dir0*, ..., *dir3* as additional columns or you must rename them accordingly in your offloading configuration.

Reviewer: ```suggestion
The following tables summarize the resulting schemas for each of the {{< product-c8y-iot >}} base collections. These schemas additionally include the virtual columns *dir0*, ..., *dir3*, which are used for internal purposes. The columns are generated during the extraction process, but neither do they have corresponding data in the operational store of {{< product-c8y-iot >}}, nor are they persisted in the data lake. Do not use *dir0*, ..., *dir3* as additional columns or rename them accordingly in your offloading configuration.
```

Author: Done... Operational Store is consistently upper-case; it's defined upfront in CDH doc

| creationTime | TIMESTAMP |
| creationTimeOffset | INTEGER |
| creationTimeWithOffset | TIMESTAMP |
| lastUpdated | TIMESTAMP |
| lastUpdatedOffset | INTEGER |
| lastUpdatedWithOffset | TIMESTAMP |
| YEAR | VARCHAR |
| MONTH | VARCHAR |
| DAY | VARCHAR |
| time | TIMESTAMP |
| timeOffset | INTEGER |
| timeWithOffset | TIMESTAMP |
| history | LIST |
| severity | VARCHAR |
| source | VARCHAR |
| status | VARCHAR |
| text | VARCHAR |
| type | VARCHAR |

{{< c8y-admon-info >}}
The column `firstOccurrenceTime` is not included in the default schema. If you want to include it in the offloading, 
it must be added manually.
{{< /c8y-admon-info >}}

> The alarms collection keeps track of alarms. An alarm may change its status over time. The alarms collection also 

Reviewer: @cheinz-sag In this section the term "view" is used a lot. To avoid confusion, what does "view" refer to in this context?

| creationTime | TIMESTAMP |
| creationTimeOffset | INTEGER |
| creationTimeWithOffset | TIMESTAMP |
| lastUpdated | TIMESTAMP |
| lastUpdatedOffset | INTEGER |
| lastUpdatedWithOffset | TIMESTAMP |
| YEAR | VARCHAR |
| MONTH | VARCHAR |
| DAY | VARCHAR |
| time | TIMESTAMP |
| timeOffset | INTEGER |
| timeWithOffset | TIMESTAMP |
| source | VARCHAR |
| text | VARCHAR |
| type | VARCHAR |

Events, just like alarms, are mutable, that is, they can be changed after their creation. Thus, the same logic as for alarms applies.

Additional views over the target table are defined in the tenant's space in Dremio. Their names are defined as target table name plus *_all*, *_latest*, and *_c8y_cdh_latest_materialized* respectively. The following examples use *events* as target table name:

> * **events_all**:  A view with  all states of all events.

Reviewer: ```suggestion
* **events_all** -  A view with all states of all events.
```

> * **events_latest**: A view containing only the latest state of all events without prior states.

Reviewer: ```suggestion
* **events_latest** - A view containing only the latest state of all events without prior states.
```

Reviewer 2: Need to be careful! "A view with  all states of all events." implies that we capture ALL state changes (Change Data Capture, CDC), but we actually might miss a few transitions.

=> A view with all CAPTURED states of all events.

(2): there's an extra space between "with" and "all" => "with__all"

Author: Done

> * **events_c8y_cdh_latest_materialized**: An optional view which materializes the **events_latest** view if the offloading configuration has view materialization enabled.

Reviewer: ```suggestion
* **events_c8y_cdh_latest_materialized** - An optional view which materializes the **events_latest** view if the offloading configuration has the view materialization enabled.
```

Author: Done

The views are provided in your Dremio space. For details on views and spaces in Dremio, see the section [Refining Offloaded Cumulocity IoT Data](/datahub/working-with-datahub/#refining-offloaded).

{{< c8y-admon-info >}}

> The *childDevices* and *childAssets* fields are not part of the default offloading columns. They were included in previous versions, but lead to problems for a high number of list items in those fields. In such a case, the columns were no more readable by Dremio. If they need to be included in the offloaded data, they can be defined as additional result columns. However, you have to ensure that the number of list items in those fields does not exceed the Dremio limit configured in your environment.

Reviewer: ```suggestion
The fields **childDevices** and **childAssets** are not part of the default offloading columns. They were included in previous versions, but lead to problems for a high number of list items in those fields. In such a case, the columns were no more readable by Dremio. If they need to be included in the offloaded data, they can be defined as additional result columns. However, you have to ensure that the number of list items in those fields does not exceed the Dremio limit configured in your environment.
```

Author: Done

| ... | C | 2.0791169082 | ... |

{{< c8y-admon-info >}}

> You should try to ensure that the data you feed into the measurements base collection is consistent. If measurements of 

Reviewer: ```suggestion
Try to ensure that the data you feed into the measurements base collection is consistent. If measurements of the same type vary in the fragment structures, the resulting target table might not have the expected schema. A common problem, for example, are varying data types of the values like one value being 2.079 and another one NaN.
```

Author: Done

> the same type vary in the fragment structures, the resulting target table might not have the expected schema. A common 

Reviewer: Combine to one entry (see suggestion above).
